framework/src/outputs/PNGTest.C

```diff
@@ -0,0 +1,279 @@
+}
+
+// Funtion for making the _mesh_function object.
+void PNGTest::makeMeshFunc() {
+
+  const std::vector<unsigned int> var_nums = {0};
+
+//  _fe_problem->addUserObject("Output", "output", meters);
+
+  // Find the values that will be used for rescaling purposes.
+  calculateRescalingValues();
+
+  // Set up the mesh_function
+  _mesh_function = libmesh_make_unique<MeshFunction>(
+      *_es_ptr, nls.serializedSolution(), nls.dofMap(), var_nums);
+  _mesh_function->init();
+  const DenseVector<Number> num(1);// = .5;
+  _mesh_function->enable_out_of_mesh_mode(num);
+}
+
+void PNGTest::calculateRescalingValues()
+{
+  // The min and max.
+  scalingMin = nls.serializedSolution().min();
+  scalingMax = nls.serializedSolution().max();
+  shiftValue = 0;
+
+  if(scalingMin < 0)
+  {
+    shiftValue += scalingMin;
+  }
+  else if(scalingMin > 0)
+  {
+    shiftValue -= scalingMin;
+  }
+
+  scalingMax += shiftValue;
+
+  // for(auto i = 0; i < nls.serializedSolution().size(); i++)
+  // {
+  //   vecx[i] = (vecx[i] + shiftValue) * (1/scalingMax);
+  // }
+
+}
+
+// Initialize by running the makeMeshFunc method.
+// void PNGTest::initialize() {
+//
+//   makeMeshFunc();
+// }
+//
+// void PNGTest::execute() {
+//
+//   makePNG();
+// }
+
+void PNGTest::output(const ExecFlagType & /*type*/)
+{
+  makeMeshFunc();
+  box = MeshTools::create_bounding_box(*_mesh_ptr);
+  makePNG();
+}
+
+void PNGTest::makePNG() {
+
+//   if(appe == "three")
+//     appe = "four";
+//
+//   else if(appe == "two")
+//     appe = "three";
+//
+//   else if(appe == "one")
+//     appe = "two";
+//
+//   std::string PNGFile = "Adam";
+//
+//   PNGFile += appe;
+//   PNGFile += ".png";
+
+  numpng++;
+
+//  BoundingBox box = MeshTools::create_bounding_box(*_mesh_ptr);
+
+  //FindBBox fbbx;
+  //std::cout << "The min and the max " << (comm().min(box.min()))(0) << comm().max(box.max()) << "\n";
+
+//  _mesh_ptr->comm(); ----------- communicator we have access to.
+//  comm(); ------------------ Another communicator we have access to.
+  //_app._comm ----- this is the communicator.  It's protected.
+
+
+
+  BoundingBox boxy = _mesh_ptr->getInflatedProcessorBoundingBox(.1);
+
+  Point ptx = box.max();
+  Point ptn = box.min();
+
+  //_mesh_ptr->comm().min(ptn);
+
+  Real distx = ptx(0) - ptn(0);
+  Real disty = ptx(1) - ptn(1);
+
+  std::cout << "Values: " << distx << " " << disty << "\n";
+
+  std::string PNGFile2 = PNGFile;
+  PNGFile2 += std::to_string((int)numpng);
+
+  FILE * fp = nullptr;
+  png_structrp pngp = nullptr;
+  png_infop infop = nullptr;
+  png_bytep row = nullptr;
+//    int status = 1;
+  Real depth = 8;
+//  Real resolution = 3000;
+  // Real xdist = 2;
+  // Real ydist = 2;
+  Real width = distx*resolution;
+  Real height = disty*resolution;
+
+  if(picked != -1) {
+    if(numpng != picked) {
+      std::cout << "Not the picked one.";
+      return;
+    }
+    std::cout << "The picked one." << PNGFile2;
+  }
+
+  else
+    std::cout << "Nothing selected.";
```

Comment:
These stream outputs need to be dropped - your object should probably be silent.